src/main/java/socialite/ui/PersonCard.java

```diff
@@ -70,19 +73,104 @@ public PersonCard(Person person, int displayedIndex) {
         phone.setText(person.getPhone().value);
         remark.setText(person.getRemark().value);
-        tiktokIcon.setImage(new Image(this.getClass().getResourceAsStream("/images/tik-tok.png")));
-        tiktok.setText(" @" + person.getTiktok().value + " ");
-        twitterIcon.setImage(new Image(this.getClass().getResourceAsStream("/images/twitter.png")));
-        twitter.setText(" @" + person.getTwitter().value + " ");
+        this.makeHandle(person.getFacebook(), Platform.FACEBOOK);
+        this.makeHandle(person.getInstagram(), Platform.INSTAGRAM);
+        this.makeHandle(person.getTelegram(), Platform.TELEGRAM);
+        this.makeHandle(person.getTiktok(), Platform.TIKTOK);
+        this.makeHandle(person.getTwitter(), Platform.TWITTER);
+
         person.getTags().stream()
                 .sorted(Comparator.comparing(tag -> tag.tagName))
                 .forEach(tag -> tags.getChildren().add(new Label(tag.tagName)));
-        facebookIcon.setImage(new Image(this.getClass().getResourceAsStream("/images/facebook.png")));
-        facebook.setText(" @" + person.getFacebook().value + " ");
-        instagramIcon.setImage(new Image(this.getClass().getResourceAsStream("/images/instagram.png")));
-        instagram.setText(" @" + person.getInstagram().value + " ");
-        telegramIcon.setImage(new Image(this.getClass().getResourceAsStream("/images/telegram.png")));
-        telegram.setText(" @" + person.getTelegram().value + " ");
+    }
+
+    private void openBrowser(String url) {
+        String os = System.getProperty("os.name").toLowerCase();
+        Runtime runtime = Runtime.getRuntime();
+        try {
+            if (os.indexOf("win") >= 0) {
+                runtime.exec("rundll32 url.dll,FileProtocolHandler " + url);
+            } else if (os.indexOf("mac") >= 0) {
+                runtime.exec("open " + url);
+            } else if (os.indexOf("nix") >= 0 || os.indexOf("nux") >= 0) {
+                runtime.exec("xdg-open " + url);
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void makeHandle(Handle handle, Platform platform) {
+        String value = handle.get();
+        switch (platform) {
+        case FACEBOOK:
+            if (value != null && !value.equals("")) {
+                this.facebook.setText(value + " ");
+                this.facebookIcon.setImage(new Image(this.getClass().getResourceAsStream("/images/facebook.png")));
+                this.facebook.setOnMouseEntered(Event -> this.facebook.setUnderline(true));
+                this.facebook.setOnMouseExited(Event -> this.facebook.setUnderline(false));
+                this.facebook.setOnMouseClicked(Event -> this.openBrowser(
+                        person.getFacebook().getUrl()));
+            } else {
+                this.facebook.setText(null);
+                this.facebookIcon.setFitWidth(0);
+            }
+            break;
+        case INSTAGRAM:
```

Comment:
This can probably be abstracted out to reduce the duplication of attaching the event handlers for each platform separately, but not very important.